### ISSUE TITLE
[Feature] use redshift_connector for metadata wherever possible

### ISSUE THREAD

**Issue by dataders** [dbt-labs/dbt-redshift#776]
### Is this your first time submitting a feature request?

- [X] I have read the [expectations for open source contributors](https://docs.getdbt.com/docs/contributing/oss-expectations)
- [X] I have searched the existing issues, and I could not find an existing issue for this feature
- [X] I am requesting a straightforward extension of existing dbt-redshift functionality, rather than a Big Idea better suited to a discussion

### Describe the feature

Wherever possible, fully rely on redshift driver for metadata and not SQL info_schema or any postgres  system view like pg_*.

This will future-proof the adapter as well as unblock robust support for Datasharing (see #742)

Exhaustive list of related issues and PRs (so far)

Throughout the process, we may discover shortcomings in the driver. As @ryanwaldof says in #742: 
> If this isn't possible because our APIs don't support your use cases lets chat so we can figure out how to fix this. If we need to provide more info in our APIs, we need to provide more info in our APIs.

In these circumstances, migration to driver for metadata would pause while the driver adds the required information.

Additionally, it is crucially important that these changes are sufficiently tested and performance is considered.

One idea that @colin-rogers-dbt and I discussed was that these metadata-related changes might be merged into a feature branch for longer-term testing with customers before merging and releasing as a new minor version of dbt-redshift.

### related issues

```[tasklist]
### Tasks
- [ ] #179 
- [ ] #452
- [ ] #554 
- [ ] #555
- [ ] #571
- [ ] #736
- [ ] #816
```

**Comment by lgwoodmark** [comment]
Is there any work being done on this? Because I would love to use exactly this setup (Aurora + federated Query + Redshift) for a customer of ours.
